io.fairyproject.modules/module.command/core-command/src/main/java/io/fairyproject/command/CommandService.java

```diff
@@ -100,8 +102,8 @@ public void preInit() {
                 .build());
     }
 
-    @PostInitialize
-    public void init() {
+    @Subscribe
+    public void init(final PostServiceInitialEvent event) {
```

Comment:
What's the reason to use PostServiceInitialEvent rather than PostInitialize?

Reply:
PostInitialize will call on server enable from most likely a different thread after it locks the mainthread, causing a deadlock on “onEnable”

Reply:
Problem with this is outlined in another branch as it won’t properly work with most stuff. Quite an issue…

Reply:
We need a replacement to PostInitialize which is mainthreaded OR which is called in a main-thread blocking context.

Reply:
I see, ever since I made ClassPathScanner threaded PreInitialize and PostInitialize are not called through main thread but main thread are blocked during these period, not quite sure the reasoning behind the deadlock tho

Reply:
Well essentially the CommandService tries to register the commands by accessing the SimplePluginManager, which in itself requires the mainthread to be non-blocked. And since it keeps awaiting for the main-thread to unlock, the main-thread stays in a blocked status (plugin not fully loaded as commands aren’t loaded, view the joinAll in the PluginService thing) and the module service thread as waiting (awaits main-thread)